ccl/multiregionccl: deflake multi-region datadriven test

kvoli opened the pull request:
Previously, `TestMultiRegionDataDriven` allowed kvserver (replicate queue) lease transfers to occur for lease count rebalancing. Load based rebalancing was disabled in an earlier patch #109050, however we saw in interleaving replication changes -- leading to flaky behavior.

Bump the `kv.allocator.min_lease_transfer_interval` to 5 minutes, effectively disabling internal lease transfers. Lease transfers which are required for lease preference satisfaction, and manual transfers are still permitted.

Fixes: #109215
Release note: None

---

`TestMultiRegionDataDriven` speeds up replication changes by proactively
enqueuing replicas into various queues. This has the benefit of reducing
the time taken after zone config changes, however the downside of added
overhead.

Disable the test under deadlock builds, as the test is slow and
susceptible to (legitimate) timing issues on a deadlock build.

Informs: https://github.com/cockroachdb/cockroach/issues/109215
Release note: None

kvoli committed:
ccl/multiregionccl: disable transfers in multiregion test

Previously, `TestMultiRegionDataDriven` allowed kvserver (replicate
queue) lease transfers to occur for lease count rebalancing. Load based
rebalancing was disabled in an earlier patch #109050, however we saw in
interleaving replication changes -- leading to flaky behavior.

Bump the `kv.allocator.min_lease_transfer_interval` to 5 minutes,
effectively disabling internal lease transfers. Lease transfers which
are required for lease preference satisfaction, and manual transfers are
still permitted.

Fixes: #109215
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/109819)
<!-- Reviewable:end -->

kvoli committed:
ccl/multiregionccl: skip multi-region dd under deadlock

`TestMultiRegionDataDriven` speeds up replication changes by proactively
enqueuing replicas into various queues. This has the benefit of reducing
the time taken after zone config changes, however the downside of added
overhead.

Disable the test under deadlock builds, as the test is slow and
susceptible to (legitimate) timing issues on a deadlock build.

Informs: #109215
Release note: None

rafiss reviewed (approved):
nice improvements, ty

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/109819#-:-NdGHEz6Cq4tyb9GbMiC:b-1duap8)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @kvoli)

___
*[`pkg/ccl/multiregionccl/datadriven_test.go` line 117 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/109819#-NdGH9I83ODa_Mz-OSSZ:-NdGH9I90TA1gLvzVsSf:b3lqna5) ([raw file](https://github.com/cockroachdb/cockroach/blob/c156fb7393679fdc04957183b8fcd21963011cef/pkg/ccl/multiregionccl/datadriven_test.go#L117)):*
> ```Go
> 
> 	skip.UnderRace(t, "flaky test")
> 	skip.UnderDeadlock(t, "flaky test")
> ```

super nit: the commit message has a nice explanation. that (or a summary) could be included here in the code so that a future reader can more easily understand why it's legitimate to skip under deadlock


<!-- Sent from Reviewable.io -->

kvoli reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/109819)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @rafiss)

___
*[`pkg/ccl/multiregionccl/datadriven_test.go` line 117 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/109819#-NdGH9I83ODa_Mz-OSSZ:-NdGNn640PbVoyYgOdFF:b-vtbaha) ([raw file](https://github.com/cockroachdb/cockroach/blob/c156fb7393679fdc04957183b8fcd21963011cef/pkg/ccl/multiregionccl/datadriven_test.go#L117)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

super nit: the commit message has a nice explanation. that (or a summary) could be included here in the code so that a future reader can more easily understand why it's legitimate to skip under deadlock
</blockquote></details>

Good idea, I've added a slightly modified commit message summary here.


<!-- Sent from Reviewable.io -->

kvoli commented:
TYFTR

bors r=rafiss
